### DISCUSSION TITLE
Dipy for VR/AR

### DISCUSSION THREAD

**Discussion by alecrimi** [dipy/dipy#2360] [1. Ideas & Feedback]
## Description
I am working on a visualizer for AR/VR

Long story short it is not so easy to manage trk file. It will be easier to extrapolate from 3d coordinates of the points in 3D for each streamlines (e.g. exported simply on a CSV file).

What is the best way to get those coordinates?

**Comment by skoudoro** [comment]
Here an example of accessing 1 streamline.  You just need to loop over it if you want to save it as csv

```python 
from dipy.io.stateful_tractogram import StatefulTractogram
from dipy.io.streamline import load_tractogram

tracto = load_tractogram('AF_L.trk', 'same', bbox_valid_check=False)
streamlines = tracto.streamlines
print(streamlines.get_data().shape)
print(streamlines._data.shape)
print(streamlines._lengths.shape)
print(streamlines._offsets.shape)
print(streamlines._offsets[:10])
# 3d coordinate of 1 streamline
first = streamlines._data[0:273]
```
